### ISSUE TITLE
Use GitHub issues and pull requests as context for copilot

### ISSUE THREAD

**Issue by danillouz** [microsoft/vscode-copilot-release#135]
Hi there!

I'm not sure if this would be (technically) feasible (currently), as I'm still learning how Copilot works. But I'm wondering if GitHub issues and pull requests (PRs) could be used as context to provide better suggestions.

Especially while debugging something, I will often end up in a GitHub issue of a library I'm using that has more helpful information or a solution.

When using Copilot, I'll often treat it as if I'm pair programming with someone. So I think it would be really useful if Copilot can either point me to relevant ticket(s)/PR(s), or propose a solution based on them (e.g. when asking about an error message).

I think it would be cool if a project's dependencies (e.g. `package.json`) could be used for this (e.g. to find relevant issues for a specific version).

And following this train of thought, I was thinking that another great source of context could be RFCs (maybe these should become "its own thing" like issues in GitHub?).

WDYT?

### Feasibility
I found this article on the engineering blog that explains a bit how Copilot contextualizes code, and why it's important:
https://github.blog/2023-05-17-how-github-copilot-is-getting-better-at-understanding-your-code/

Where it does mention:
> Right now, transformers that are fast enough to power GitHub Copilot can process about 6,000 characters at a time.

But also:
> The first version was only able to consider the file you were working on in your IDE to be contextually relevant. But we knew context went beyond that. Now, just a year later, we’re experimenting with algorithms that will consider your entire codebase to generate customized suggestions.

So this hints at that it might be possible to pull in a lot more data for context?

### Other (potentially) relevant tickets
I found some other tickets that also mention improving context, but I believe none mention issues(s)/PR(s) explicitly:
- https://github.com/microsoft/vscode-copilot-release/issues/32
- https://github.com/microsoft/vscode-copilot-release/issues/29
- https://github.com/microsoft/vscode-copilot-release/issues/25
- https://github.com/microsoft/vscode-copilot-release/issues/14

**Comment by rbleattler** [comment]
Any updates on this? I feel it could be very helpful, especially for organizations.

**Comment by clipperhouse** [comment]
I’d like to see this, specifically the PR as selectable context — the files of the PR, as well as comments of the PR, and follow links to related issues to bring them in context.

I would want to prompt with:

- Please review this PR
- Is this PR idiomatic compared to the rest of the codebase?
- How does this PR compare to other PRs?

**Comment by lkraav** [comment]
Yep. Today I started looking for a way to add PR as context due to https://github.com/githubnext/copilot-workspace-user-manual/issues/68 failing, and ended up here.

Considering squashing PR into a single diff and feeding that.
